server/service/team_policies.go

```diff
@@ -366,13 +366,17 @@ func (svc *Service) modifyPolicy(ctx context.Context, teamID *uint, id uint, p f
 		})
 	}
 
+	var shouldRemoveAll bool
 	if p.Name != nil {
 		policy.Name = *p.Name
 	}
 	if p.Description != nil {
 		policy.Description = *p.Description
 	}
 	if p.Query != nil {
+		if policy.Query != *p.Query {
```

Comment:
Let's add an integration test on `integration_core_test.go`.

Reply:
This test looks like it mostly covers this case? https://github.com/fleetdm/fleet/blob/13879-policy-counts/server/service/integration_core_test.go#L1820-L1881
I went ahead and added an additional assertion that the failing list is also empty tho.

Reply:
I believe such test is missing to check the values for `mgpResp.Policy.PassingHostCount`/`mgpResp.Policy.FailingHostCount` after modifying the SQL of the policy.

Reply:
done!

Reply:
Ah, but you need to generate results before the modify call, you can use:
```go
err = s.ds.RecordPolicyQueryExecutions(context.Background(), h1.Host, map[uint]*bool{policiesResponse.Policies[0].ID: ptr.Bool(true)}, time.Now(), false)
require.NoError(t, err)
```

Reply:
Oh I see what you mean, idk why I thought there would already be results at this point in the test. fixing

Reply:
@lucasmrod that made me realize that I wasn't zeroing out the counts in the response when modifying the policy. Fixed that and moved the test to where it makes more sense and will verify that there was actually a change.